### PR TITLE
Support Reports Field

### PR DESCRIPTION
**Pull request by nhakmiller**
### Background

We're going to be adding support soon for a reports field, which will help report generation. Since we want to move some information currently encoded in tags to the reports field, this change just allows the field to be present and not throw an error during testing. The Panther backend throws away the Reports field when processing the configuration.

### Changes

* Allow a reports field to exists, which will be a map of strings (report names, which will also later be treated as tags) to objects (typically either a list of strings or a dictionary)

### Testing

* Added some different configurations of report fields to some policies I'm working on locally and tested and uploaded them, no issues

### CODE REVIEW

File: panther_analysis_tool/main.py [hunk 1]
```diff
@@ -80,12 +80,15 @@ def get(self, arg: str, default: Any = None) -> Any:
             str,
         Optional('Suppressions'): [str],
         Optional('Tags'): [str],
+        Optional('Reports'): {
```
Comment (jacknagz) [inline panther_analysis_tool/main.py]: Can we also fix the PolicyID vs RuleID issues in the schema?

Comment (jacknagz) [reply panther_analysis_tool/main.py]: ResourceType vs LogType

Comment (nhakmiller) [reply panther_analysis_tool/main.py]: That's really more of a backend Panther change than a change the `panther_analysis_tool` needs, the backend currently doesn't have a concept of RuleID or LogType so we could let the `panther_analysis_tool` support these fields, but anyone who actually used them wouldn't be able to upload those rules to Panther.

**Comment by kostaspap** [comment]
Something I just realized, we need to add support for the `DedupPeriodMinutes` field. Right now it is default to 60 minutes, but people should be able to change it eventually. They can do it from the UI with https://github.com/panther-labs/panther/pull/475, so we should eventually let them configure it from the analysis tool as well!

**Pull request merged**